crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -182,6 +188,188 @@ impl TransactionBlock {
+                    query = query.filter(tx::dsl::checkpoint_sequence_number.lt(*c as i64));
+                }
+
+                if let Some(a) = &filter.sign_address {
+                    let sub_query = tx_senders::dsl::tx_senders
+                        .select(tx_senders::dsl::tx_sequence_number)
+                        .filter(tx_senders::dsl::sender.eq(a.into_vec()));
+                    query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
+                }
+
+                if let Some(a) = &filter.recv_address {
+                    let sub_query = tx_recipients::dsl::tx_recipients
+                        .select(tx_recipients::dsl::tx_sequence_number)
+                        .filter(tx_recipients::dsl::recipient.eq(a.into_vec()));
+                    query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
+                }
+
+                if let Some(o) = &filter.input_object {
+                    let sub_query = tx_input_objects::dsl::tx_input_objects
+                        .select(tx_input_objects::dsl::tx_sequence_number)
+                        .filter(tx_input_objects::dsl::object_id.eq(o.into_vec()));
+                    query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
+                }
+
+                if let Some(o) = &filter.changed_object {
+                    let sub_query = tx_changed_objects::dsl::tx_changed_objects
+                        .select(tx_changed_objects::dsl::tx_sequence_number)
+                        .filter(tx_changed_objects::dsl::object_id.eq(o.into_vec()));
+                    query = query.filter(tx::dsl::tx_sequence_number.eq_any(sub_query));
+                }
+
+                if let Some(txs) = &filter.transaction_ids {
+                    let digests: Vec<_> = txs.iter().map(|d| d.to_vec()).collect();
+                    query = query.filter(tx::dsl::transaction_digest.eq_any(digests));
+                }
+
+                query
+            })
+            .await?;
+
+        let mut conn = Connection::new(prev, next);
+
+        for stored in results {
+            let cursor = Cursor::new(stored.cursor()).encode_cursor();
+            let transaction = TransactionBlock::try_from(stored)?;
+            conn.edges.push(Edge::new(cursor, transaction));
+        }
+
+        Ok(conn)
+    }
+}
+
+impl TransactionBlockFilter {
+    pub(crate) fn merge(self, other: Self) -> Option<Self> {
+        fn by_eq<T: Eq>(a: T, b: T) -> Option<T> {
+            (a == b).then_some(a)
+        }
+
+        fn by_max<T: Ord>(a: T, b: T) -> Option<T> {
+            Some(a.max(b))
+        }
+
+        fn by_min<T: Ord>(a: T, b: T) -> Option<T> {
+            Some(a.min(b))
+        }
+
+        macro_rules! merge {
+            ($field:ident, $body:expr) => {
+                merge_filter(self.$field, other.$field, $body)
+            };
+        }
+
+        Some(Self {
+            function: merge!(function, by_eq)?,
```

Comment:
Noticed that we should be doing something more sophisticated than an equality check here.  That's a non-trivial piece of work, so I'm going to handle that in a follow-up because we don't currently merge filters that might have two `function`s set.

Reply:
I see, thanks.